### PR TITLE
Add surface graph snapshot tests for datalog.

### PR DESCRIPTION
**Pull request by tylerhou**

### CODE REVIEW

**Review by shadaj — CHANGES_REQUESTED** [review]
Looks like clippy is failing due to an unnecessary `return`.

**Comment by tylerhou** [comment]
> Looks like clippy is failing due to an unnecessary `return`.

Done.

File: hydroflow_datalog/src/lib.rs [hunk 1]
```diff
@@ -243,23 +248,34 @@ mod tests {
...
         ($program:literal) => {
-            let out = &gen_datalog_program(parse_quote!($program), quote::quote! { hydroflow });
+            let graph = gen_hydroflow_graph(parse_quote!($program));
+
+            insta::with_settings!({snapshot_suffix => "surface_graph"}, {
+                insta::assert_display_snapshot!(graph.surface_syntax_string());
+            });
 
+            // Have to make a new graph as the above closure borrows.
+            let graph2 = gen_hydroflow_graph(parse_quote!($program));
```
Comment (shadaj) [inline hydroflow_datalog/src/lib.rs:260]: Can we just clone the graph before it's grabbed by the closure?

Comment (tylerhou) [reply hydroflow_datalog/src/lib.rs:260]: FlatGraph doesn't implement Clone. I tried Deriving it but Node does not implement Clone.

**Review by shadaj — APPROVED** [review]
LGTM! This is a _way_ nicer to check compile outputs.

**Pull request merged**